apps/web/src/components/WorkEmailCard.tsx/WipeWorkEmailDialog.tsx

```diff
@@ -0,0 +1,119 @@
+import { useState } from "react";
+import { useIntl } from "react-intl";
+import { FormProvider, useForm } from "react-hook-form";
+import { useMutation } from "urql";
+
+import { Dialog, Button } from "@gc-digital-talent/ui";
+import { commonMessages } from "@gc-digital-talent/i18n";
+import { graphql } from "@gc-digital-talent/graphql";
+import { toast } from "@gc-digital-talent/toast";
+
+const WipeWorkEmail_Mutation = graphql(/* GraphQL */ `
+  mutation WipeWorkEmail($id: UUID!) {
+    wipeUserWorkEmail(id: $id) {
+      id
+    }
+  }
+`);
+
+interface WipeWorkEmailDialogProps {
+  id: string;
+  workEmail: string;
+}
+
+const WipeWorkEmailDialog = ({ id, workEmail }: WipeWorkEmailDialogProps) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const intl = useIntl();
+
+  const methods = useForm();
+  const { handleSubmit } = methods;
+
+  const dialogTitle = intl.formatMessage({
+    defaultMessage: "Remove this work email?",
+    id: "PTfLmk",
+    description: "Dialog header",
+  });
+  const handleError = () => {
+    toast.error(
+      intl.formatMessage({
+        defaultMessage: "Error: failed to delete",
+        id: "jyECqY",
+        description: "Error for failing to delete",
+      }),
+    );
+  };
+
+  const [{ fetching }, executeWipeWorkInformationMutation] = useMutation(
+    WipeWorkEmail_Mutation,
+  );
+  const wipeWorkInformation = async () => {
+    await executeWipeWorkInformationMutation({ id })
+      .then((result) => {
+        if (result.data?.wipeUserWorkEmail) {
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "Work email deleted",
+              id: "3bVNFv",
+              description: "Successfully cleared work email information",
+            }),
+          );
```

Comment:
Sort of nit-pciky but, we use the term "Remove" everywhere else so maybe we shiuld replace "deleted" in this and the error message with "removed".

Also, I tihnk we use periods in our toasts.

@NienkeBr Do you have any strong opinions about this?

Reply:
Aye aye, see https://github.com/GCTC-NTGC/gc-digital-talent/pull/15149/commits/dd641ed11f5106938afed5a01577cd227d5b36a2